projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/JobExecutionFilterSpec.java

```diff
@@ -41,6 +41,19 @@ public Predicate toPredicate(Root<DataJobExecution> root, CriteriaQuery<?> query
          if (CollectionUtils.isNotEmpty(filter.getStatusIn())) {
             predicates.add(root.get(DataJobExecution_.STATUS).in(filter.getStatusIn()));
          }
+
+         if (filter.getJobNameIn() != null && filter.getJobNameIn().size() > 0) {
+            predicates.add(root.get(DataJobExecution_.DATA_JOB).get(DataJob_.NAME).in(filter.getJobNameIn()));
+         }
+
+         if (filter.getStartTimeLte() != null) {
```

Comment:
It will be good if you arrange the filters in the following order:
StartTimeLte
StartTimeGte
EndTimeLte
EndTimeGte
StatusIn
JobNameIn

This applies to all files.